### PR TITLE
unit: make UNIT() cast function deal with NULL pointers

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #10681

### CODE REVIEW

**Comment by poettering** [comment]
An alternative to #10684's first patch

File: src/core/unit.h [hunk 1]
```diff
@@ -582,7 +582,12 @@ extern const UnitVTable * const unit_vtable[_UNIT_TYPE_MAX];
         }
 
 /* For casting the various unit types into a unit */
-#define UNIT(u) (&(u)->meta)
+#define UNIT(u)                                         \
+        ({                                              \
+                typeof(u) _u_ = (u);                    \
+                Unit *w = _u_ ? &(_u_)->meta : NULL;    \
+                w;                                      \
```
Comment (yuwata) [inline src/core/unit.h]: Maybe it is safer to use `_w_`.

Comment (poettering) [reply src/core/unit.h]: indeed

**Comment by poettering** [comment]
pushed with @yuwata's suggestion

**Comment by keszybz** [comment]
I don't think the easy-review flag makes much sense. This is **not** an easy review, macros are complicated. If you're trying to say that it's a **short** review, OK, I can live with that, but I don't think we need a label for it.

**Comment by poettering** [comment]
@keszybz i renamed the label to "quick-review" now. The reason I wanted to mark PRs with this is to indicate in the list view that something is just a few lines of changes, in contrast to 15patch mega PRs, so that drive-by reviewers maybe can have a look at those without having to make it an hour-long review-fest, if you you understand what i mean

**Comment by keszybz** [comment]
So a nicer solution would be for github to just say how many patches are in the PRs in the listing. But I guess they don't want to do this because they prefer to workflow where fixup commits are pushed on top, so the number of commits is "irrelevant".

**Comment by keszybz** [comment]
OK, two builds passed, let's merge.

**Pull request merged**